src/global/helpers/native-date-adapter.ts

```diff
@@ -0,0 +1,224 @@
+    return (
+      (NativeDateAdapter.DAYS_PER_WEEK +
+        this.getDayOfWeek(firstOfMonth) -
+        this.getFirstDayOfWeek()) %
+      NativeDateAdapter.DAYS_PER_WEEK
+    );
+  }
+
+  /** Gets the year of the input date. */
+  public getYear(date: Date): number {
+    return date.getFullYear();
+  }
+
+  /** Gets the month of the input date. */
+  public getMonth(date: Date): number {
+    return date.getMonth();
+  }
+
+  /** Gets the day of the input date. */
+  public getDate(date: Date): number {
+    return date.getDate();
+  }
+
+  /** Gets the day of the week of the input date. */
+  public getDayOfWeek(date: Date): number {
+    return date.getDay();
+  }
+
+  /** Gets the date in the local format. */
+  public getAccessibilityFormatDate(date: Date | string): string {
+    const options: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+    };
+    return new Intl.DateTimeFormat(documentLanguage(), options).format(new Date(date));
+  }
+
+  /**
+   * Creates an array of strings with length = 12, filled with the name of the months starting from January, in the document language.
+   * @param style See `month` in `DateTimeFormatOptions`: `long` for full name, `short` for short name, `narrow` for single letter,
+   * E.g., with January in en-gb: `long` returns "January", `short` returns "Jan", `narrow` returns "J".
+   */
+  public getMonthNames(style: 'long' | 'short' | 'narrow'): string[] {
+    const formatter = new Intl.DateTimeFormat(documentLanguage(), { month: style });
+    return this._range(12, (i) => formatter.format(new Date(2017, i, 1)));
+  }
+
+  /** Creates a string array with length = 31, filled with the days in a month, starting from 1. */
+  public getDateNames(): string[] {
+    const formatter = new Intl.DateTimeFormat(documentLanguage(), { day: 'numeric' });
+    return this._range(31, (i) => formatter.format(new Date(2017, 0, i + 1)));
+  }
+
+  /**
+   * Creates a string array with length = 7, filled with the name of the days in a week starting from Sunday, in the document language.
+   * @param style See `weekday` in `DateTimeFormatOptions` - 'long' for full name, 'short' for short name, 'narrow' for single letter;
+   * E.g., with Monday in en-gb: `long` returns "Monday", `short` returns "Mon", `narrow` returns "M".
+   */
+  public getDayOfWeekNames(style: 'long' | 'short' | 'narrow'): string[] {
+    const formatter = new Intl.DateTimeFormat(documentLanguage(), { weekday: style });
+    return this._range(7, (i) => formatter.format(new Date(2017, 0, i + 1)));
+  }
+
+  /* Defines which is the first day of the week (0: sunday; 1: monday; etc.). */
+  public getFirstDayOfWeek(): number {
+    return 1;
+  }
+
+  /** Calculates the number of days in a month given the year and the month. */
+  public getNumDaysInMonth(year: number, month: number): number {
+    return new Date(year, month + 1, 0, 0, 0, 0, 0).getDate();
```

Comment:
surprised that this works

Reply:
Well, while being a little counterintuitive it's a fairly solid logic, since that the day 0 of a month is always the last day of the previous month, which corresponds to its length 😅